### ISSUE TITLE
 error: ‘copy_n’ is not a member of ‘std’; did you mean ‘copy’?

### ISSUE THREAD

**Issue by goldyfruit** [hyprwm/xdg-desktop-portal-hyprland#267]
I'm running openSUSE Tumbleweed with GCC 14.2.
```
┌─[goldyfruit@x270] - [~/Development/xdg-desktop-portal-hyprland] - [Tue Sep 24, 11:16]
└─[$] <git:(master)> cmake --build build                                                           

[  2%] Building CXX object subprojects/sdbus-cpp/CMakeFiles/sdbus-c++-objlib.dir/src/Connection.cpp.o
In file included from /home/goldyfruit/Development/xdg-desktop-portal-hyprland/subprojects/sdbus-cpp/src/Connection.h:31,
                 from /home/goldyfruit/Development/xdg-desktop-portal-hyprland/subprojects/sdbus-cpp/src/Connection.cpp:27:
/home/goldyfruit/Development/xdg-desktop-portal-hyprland/subprojects/sdbus-cpp/include/sdbus-c++/Message.h: In member function ‘void sdbus::Message::deserializeArrayFast(_Array&)’:
/home/goldyfruit/Development/xdg-desktop-portal-hyprland/subprojects/sdbus-cpp/include/sdbus-c++/Message.h:490:14: error: ‘copy_n’ is not a member of ‘std’; did you mean ‘copy’?
  490 |         std::copy_n(arrayPtr, elementsInMsg, items.begin());
      |              ^~~~~~
      |              copy
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-address-of-temporary’ may have been intended to silence earlier diagnostics
gmake[2]: *** [subprojects/sdbus-cpp/CMakeFiles/sdbus-c++-objlib.dir/build.make:76: subprojects/sdbus-cpp/CMakeFiles/sdbus-c++-objlib.dir/src/Connection.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:243: subprojects/sdbus-cpp/CMakeFiles/sdbus-c++-objlib.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```
Not sure if its a compiler issue or something else.

**Comment by vaxerski** [comment]
something's up with sdbus-cpp. We use a quite outdated version, maybe I should finally update.

**Comment by vaxerski** [comment]
IIRC we use 1.3

**Comment by goldyfruit** [comment]
> IIRC we use 1.3 tho 1.6 should work too?

Except if a param/feature/option got deprecated/removed I guess.

**Comment by Masber** [comment]
Hi, I am having the same issue, using Ubuntu 24.10

**Comment by Masber** [comment]
This solved this issue for me in Ubuntu 24.10

```sh
sudo apt install libsdbus-c++-bin libsdbus-c++-dev libsdbus-c++1 libsdbus-c++-doc
```

**Comment by IlSerry** [comment]
Ubuntu 24.10 with same problem here: building using libsdbus-c++* provided by package manager throwing the `copy_n` error.

I uninstalled libsdbsud-c++* and built it from git repo (https://github.com/Kistler-Group/sdbus-cpp) with no additional flags, after that i try to rebuild xdg-desktop-portal-hyprland and all work correctly!
